### PR TITLE
Add user setting to hide birthdays from deceased contacts

### LINKED ISSUE #24742: Birthdays continue to be shown in calendar after death
### Describe the bug

One of my contacts recently died, so I went and set the date of death field in their entry. However, there are still ongoing calendar entries for this contact's birthday after their date of death, and I get notifications for them.

I'm not sure if this is a contacts bug or a calendar bug, but I think it is an issue with contacts because the calendar is, as far as I'm aware, simply displaying entries populated by the contacts app.

**To Reproduce**

1. Add a contact.
2. Set date of birth.
3. Set date of death.
4. View calendar and jump to what would be the next birthday after the date of death - entry is still there.

**Expected behavior**

No more birthday entries in the calendar after the date of death.

**Actual behavior**

Birthday entries continue indefinitely.

### Server configuration

**Operating system**: Linux 5.4.0-58-generic nextcloud/contacts#64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020 x86_64

**Web server:** nginx/1.18.0 (fpm-fcgi)

**Database:** mysql 10.3.25 (reported as mysql in Nextcloud but it is actually MariaDB).

**PHP version:** 7.4.3
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, sodium, cgi-fcgi, mysqlnd, PDO, xml, bcmath, bz2, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, imagick, intl, json, exif, mysqli, pdo_mysql, Phar, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 20.0.4 - 20.0.4.0

**Contacts version:** 3.4.2

**Updated from an older Nextcloud or fresh install:** Updated

**Signing status:**

No errors have been found.

**List of activated apps:**

```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.4
  - admin_audit: 1.10.0
  - calendar: 2.1.2
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.2
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.2
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.1
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - issuetemplate: 0.7.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - nextcloud_announcements: 1.9.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.1
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - user_status: 1.0.1
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - encryption
  - files_external
  - user_ldap
```

**Nextcloud configuration:**

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.4.0",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "trashbin_retention_obligation": "auto, 30",
        "mysql.utf8mb4": true,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    },
    "apps": {
        "accessibility": {
            "enabled": "yes",
            "installed_version": "1.6.0",
            "types": ""
        },
        "activity": {
            "enabled": "yes",
            "installed_version": "2.13.4",
            "types": "filesystem"
        },
        "admin_audit": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "logging"
        },
        "backgroundjob": {
            "lastjob": "976"
        },
        "calendar": {
            "enabled": "yes",
            "installed_version": "2.1.2",
            "types": ""
        },
        "cloud_federation_api": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "filesystem"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "logging"
        },
        "contacts": {
            "enabled": "yes",
            "installed_version": "3.4.2",
            "types": "dav"
        },
        "contactsinteraction": {
            "enabled": "yes",
            "installed_version": "1.1.0",
            "types": "dav"
        },
        "core": {
            "backgroundjobs_mode": "cron",
            "enterpriseLogoChecked": "yes",
            "installedat": "1538569858.8053",
            "lastcron": "1608197103",
            "lastupdateResult": "{\"version\":\"20.0.4.0\",\"versionstring\":\"Nextcloud 20.0.4\",\"url\":\"https:\\\/\\\/download.nextcloud.com\\\/server\\\/releases\\\/nextcloud-20.0.4.zip\",\"web\":\"https:\\\/\\\/docs.nextcloud.com\\\/server\\\/20\\\/admin_manual\\\/maintenance\\\/upgrade.html\",\"changes\":\"https:\\\/\\\/updates.nextcloud.com\\\/changelog_server\\\/?version=20.0.4\",\"autoupdater\":\"1\",\"eol\":\"0\"}",
            "lastupdatedat": "0",
            "moveavatarsdone": "yes",
            "oc.integritycheck.checker": "[]",
            "previewsCleanedUp": "1",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "scss.variables": "db81cddf52fdb3c8ca1e4c859e214124",
            "shareapi_allow_group_sharing": "no",
            "shareapi_allow_public_upload": "no",
            "shareapi_allow_share_dialog_user_enumeration": "no",
            "shareapi_default_permission_cancreate": "no",
            "shareapi_default_permission_candelete": "no",
            "shareapi_default_permission_canshare": "no",
            "shareapi_default_permission_canupdate": "no",
            "shareapi_default_permissions": "1",
            "shareapi_enabled": "no",
            "theming.variables": "b92d206521717ac032f8aa58d3c7ff2f",
            "updater.secret.created": "1608143427",
            "vendor": "nextcloud"
        },
        "dashboard": {
            "enabled": "yes",
            "installed_version": "7.0.0",
            "types": ""
        },
        "dav": {
            "buildCalendarReminderIndex": "yes",
            "buildCalendarSearchIndex": "yes",
            "chunks_migrated": "1",
            "enabled": "yes",
            "installed_version": "1.16.2",
            "regeneratedBirthdayCalendarsForYearFix": "yes",
            "sendEventRemindersPush": "yes",
            "sendInvitations": "no",
            "types": "filesystem"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.10.2",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "1.10.1",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "enabled": "yes",
            "installed_version": "1.15.0",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "2.0.1",
            "types": ""
        },
        "files_rightclick": {
            "enabled": "yes",
            "installed_version": "0.17.0",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "incoming_server2server_share_enabled": "no",
            "installed_version": "1.12.1",
            "lookupServerUploadEnabled": "no",
            "outgoing_server2server_share_enabled": "no",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "no",
            "installed_version": "2.8.0",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.10.1",
            "types": "filesystem,dav"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.13.0",
            "types": "filesystem,dav"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "yes",
            "installed_version": "2.9.0",
            "types": "logging"
        },
        "gallery": {
            "enabled": "no",
            "installed_version": "18.4.0",
            "types": ""
        },
        "issuetemplate": {
            "enabled": "yes",
            "installed_version": "0.7.0",
            "types": ""
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.5.0",
            "relativedates": "1",
            "types": ""
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "authentication"
        },
        "nextcloud_announcements": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "pub_date": "Thu, 24 Oct 2019 00:00:00 +0200",
            "types": "logging"
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "2.8.0",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": "authentication"
        },
        "password_policy": {
            "enabled": "yes",
            "installed_version": "1.10.1",
            "types": "authentication"
        },
        "photos": {
            "enabled": "yes",
            "installed_version": "1.2.1",
            "types": ""
        },
        "privacy": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "prevent_group_restriction"
        },
        "recommendations": {
            "enabled": "yes",
            "installed_version": "0.8.0",
            "types": ""
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": ""
        },
        "settings": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "sendpasswordmail": "no",
            "types": "filesystem"
        },
        "support": {
            "SwitchUpdaterServerHasRun": "yes",
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "session"
        },
        "survey_client": {
            "enabled": "yes",
            "installed_version": "1.8.0",
            "types": ""
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "logging"
        },
        "text": {
            "enabled": "yes",
            "installed_version": "3.1.0",
            "types": "dav"
        },
        "theming": {
            "enabled": "yes",
            "installed_version": "1.11.0",
            "types": "logging"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": ""
        },
        "updatenotification": {
            "calendar": "2.0.3",
            "contacts": "3.4.2",
            "core": "20.0.4.0",
            "enabled": "yes",
            "files_rightclick": "0.15.1",
            "installed_version": "1.10.0",
            "types": "",
            "update_check_errors": "0"
        },
        "user_status": {
            "enabled": "yes",
            "installed_version": "1.0.1",
            "types": ""
        },
        "viewer": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": ""
        },
        "weather_status": {
            "enabled": "yes",
            "installed_version": "1.0.0",
            "types": ""
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "2.2.0",
            "types": "filesystem"
        }
    }
}
```

### Client configuration

I don't think the problem is with my client configuration, as the issue is the calendar being populated on the server. I have tried multiple browsers and the same problem occurs in all of them.

**Browser:** Firefox 84.0

**Operating system:** Ubuntu 20.04

**CardDAV-clients:** n/a

### PR DESCRIPTION
**Pull request by tcitworld**
* Resolves: #24742

## TODO

- [ ] Tests

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/appinfo/info.xml [hunk 1]
```diff
@@ -62,6 +62,7 @@
 	<settings>
 		<admin>OCA\DAV\Settings\CalDAVSettings</admin>
 		<personal>OCA\DAV\Settings\AvailabilitySettings</personal>
+		<personal>OCA\DAV\Settings\PersonalCalendarSettings</personal>
```
Comment (ChristophWurst) [inline apps/dav/appinfo/info.xml:65]: https://github.com/nextcloud/server/pull/34626#issuecomment-1280435061 you were right

**Comment by tcitworld** [comment]
I'm guessing this kind of setting makes more sense in Calendar anyway? @nextcloud/designers 
Context: https://github.com/nextcloud/server/issues/24742

**Comment by skjnldsv** [comment]
Probably yes, https://github.com/nextcloud/calendar/blob/3288a29eadab24cd8341b550c6ad6ccb0e23afd3/src/components/AppNavigation/Settings.vue#L11-L16

**Comment by jancborchardt** [comment]
@tcitworld if it would be a general setting like "Show birthdays after death" then yeah, Calendar makes sense.
However it’s a bit weirdly specific, and also you might want to see the birthday of people close to you still, while not for others.

So since this general setting would rely on the "Death date" being set in the contact anyway, this should ideally be a per-contact setting in contacts. When "Death date" is set, there can be a checkbox "Show birthday in calendar" below.

What do you think?

**Comment by tcitworld** [comment]
> So since this general setting would rely on the "Death date" being set in the contact anyway, this should ideally be a per-contact setting in contacts. When "Death date" is set, there can be a checkbox "Show birthday in calendar" below.

@jancborchardt We already have the feature to exclude specific contacts from the birthday calendar. This is about having a global setting to avoid setting this for all deceased contacts manually. It may not be needed, and we can just close #24742.
image

**Comment by jancborchardt** [comment]
@tcitworld thanks! Continued at https://github.com/nextcloud/server/issues/24742#issuecomment-2348817526

**Pull request closed without merging**